### PR TITLE
migrate  cmd/kube-proxy/app logs to structured logging

### PR DESCRIPTION
**Pull request by yxxhero**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
part of kubernetes/enhancements#1602

**Which issue(s) this PR fixes**:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
migratecmd/kube-proxy/app logs to structured logging
```

### CODE REVIEW

**Comment by Cweiping** [comment]
Thank you for your contribution.
/ok-to-test

**Comment by yxxhero** [comment]
@Cweiping  The test is all ok. what should I do next?  Thanks very much.

**Comment by Cweiping** [comment]
> @Cweiping The test is all ok. what should I do next? Thanks very much.

Nothing, thx!
/lgtm
/assign @bowei

**Comment by yxxhero** [comment]
Looking forward to your review. Thanks very much @bowei

**Comment by yxxhero** [comment]
@Cweiping  I rebase it. Please review. Thanks very much.

**Comment by yxxhero** [comment]
Looking forward to your review. Thanks very much @MrHohn @rramkumar1 @bowei

**Comment by yxxhero** [comment]
ping
/assign @bowei

**Comment by yxxhero** [comment]
any update?

**Comment by yxxhero** [comment]
@shubheksha @ehashman what I shuold do next? Thanks very much!

File: cmd/kube-proxy/app/conntrack.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func (realConntracker) setIntSysCtl(name string, value int) error {
 
 	sys := sysctl.New()
 	if val, _ := sys.GetSysctl(entry); val != value {
-		klog.Infof("Set sysctl '%v' to %v", entry, value)
+		klog.InfoS("Set sysctl", entry, value)
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/conntrack.go:100]:
```suggestion
		klog.InfoS("Set sysctl", "entry", entry, "value", value)
```

File: cmd/kube-proxy/app/conntrack.go [hunk 2]
```diff
@@ -112,7 +112,7 @@ func isSysFSWritable() (bool, error) {
 	m := mount.New("" /* default mount path */)
 	mountPoints, err := m.List()
 	if err != nil {
-		klog.Errorf("failed to list mount points: %v", err)
+		klog.ErrorS(err, "failed to list mount points")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/conntrack.go:115]:
```suggestion
		klog.ErrorS(err, "Failed to list mount points")
```

File: cmd/kube-proxy/app/conntrack.go [hunk 3]
```diff
@@ -124,8 +124,7 @@ func isSysFSWritable() (bool, error) {
 		if len(mountPoint.Opts) > 0 && mountPoint.Opts[0] == permWritable {
 			return true, nil
 		}
-		klog.Errorf("sysfs is not writable: %+v (mount options are %v)",
-			mountPoint, mountPoint.Opts)
+		klog.ErrorS(nil, "sysfs is not writable", "mountPoint", mountPoint, "mountOptions", mountPoint.Opts)
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/conntrack.go:127]:
```suggestion
		klog.ErrorS(nil, "Sysfs is not writable", "mountPoint", mountPoint, "mountOptions", mountPoint.Opts)
```

File: cmd/kube-proxy/app/server.go [hunk 4]
```diff
@@ -429,7 +429,7 @@ func (o *Options) loadConfig(data []byte) (*kubeproxyconfig.KubeProxyConfigurati
 		}
 
 		// Continue with the v1alpha1 object that was decoded leniently, but emit a warning.
-		klog.Warningf("using lenient decoding as strict decoding failed: %v", err)
+		klog.InfoS("using lenient decoding as strict decoding failed", "reason", err)
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:432]:
```suggestion
		klog.InfoS("Using lenient decoding as strict decoding failed", "err", err)
```

File: cmd/kube-proxy/app/server.go [hunk 5]
```diff
@@ -476,14 +476,17 @@ with the apiserver API to configure the proxy.`,
 			cliflag.PrintFlags(cmd.Flags())
 
 			if err := initForOS(opts.WindowsService); err != nil {
-				klog.Fatalf("failed OS init: %v", err)
+				klog.ErrorS(err, "failed OS init")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:479]:
```suggestion
				klog.ErrorS(err, "Failed OS init")
```

File: cmd/kube-proxy/app/server.go [hunk 6]
```diff
@@ -476,14 +476,17 @@ with the apiserver API to configure the proxy.`,
...
+				os.Exit(255)
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:480]: os.Exit(1)  ?

File: cmd/kube-proxy/app/server.go [hunk 7]
```diff
@@ -476,14 +476,17 @@ with the apiserver API to configure the proxy.`,
...
 			}
 
 			if err := opts.Complete(); err != nil {
-				klog.Fatalf("failed complete: %v", err)
+				klog.ErrorS(err, "failed complete")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:484]:
```suggestion
				klog.ErrorS(err, "Failed complete")
```

File: cmd/kube-proxy/app/server.go [hunk 8]
```diff
@@ -476,14 +476,17 @@ with the apiserver API to configure the proxy.`,
...
+				os.Exit(255)
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:485]: os.Exit(1) ?

File: cmd/kube-proxy/app/server.go [hunk 9]
```diff
@@ -476,14 +476,17 @@ with the apiserver API to configure the proxy.`,
...
 			}
 			if err := opts.Validate(); err != nil {
-				klog.Fatalf("failed validate: %v", err)
+				klog.ErrorS(err, "failed validate")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:488]:
```suggestion
				klog.ErrorS(err, "Failed validate")
```

File: cmd/kube-proxy/app/server.go [hunk 10]
```diff
@@ -476,14 +476,17 @@ with the apiserver API to configure the proxy.`,
...
+				os.Exit(255)
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:489]: os.Exit(1) ?

File: cmd/kube-proxy/app/server.go [hunk 11]
```diff
@@ -503,7 +506,8 @@ with the apiserver API to configure the proxy.`,
 	var err error
 	opts.config, err = opts.ApplyDefaults(opts.config)
 	if err != nil {
-		klog.Fatalf("unable to create flag defaults: %v", err)
+		klog.ErrorS(err, "unable to create flag defaults")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:509]:
```suggestion
		klog.ErrorS(err, "Unable to create flag defaults")
```

File: cmd/kube-proxy/app/server.go [hunk 12]
```diff
@@ -503,7 +506,8 @@ with the apiserver API to configure the proxy.`,
...
+		os.Exit(255)
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:510]: os.Exit(1) ?

File: cmd/kube-proxy/app/server.go [hunk 13]
```diff
@@ -585,15 +589,15 @@ func serveHealthz(hz healthcheck.ProxierHealthUpdater, errCh chan error) {
 	fn := func() {
 		err := hz.Run()
 		if err != nil {
-			klog.Errorf("healthz server failed: %v", err)
+			klog.ErrorS(err, "healthz server failed")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:592]:
```suggestion
			klog.ErrorS(err, "Healthz server failed")
```

File: cmd/kube-proxy/app/server.go [hunk 14]
```diff
@@ -585,15 +589,15 @@ func serveHealthz(hz healthcheck.ProxierHealthUpdater, errCh chan error) {
...
 			if errCh != nil {
 				errCh <- fmt.Errorf("healthz server failed: %v", err)
 				// if in hardfail mode, never retry again
 				blockCh := make(chan error)
 				<-blockCh
 			}
 		} else {
-			klog.Errorf("healthz server returned without error")
+			klog.ErrorS(nil, "healthz server returned without error")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:600]:
```suggestion
			klog.ErrorS(nil, "Healthz server returned without error")
```

File: cmd/kube-proxy/app/server.go [hunk 15]
```diff
@@ -641,14 +645,14 @@ func serveMetrics(bindAddress, proxyMode string, enableProfiling bool, errCh cha
...
-	klog.Infof("Version: %+v", version.Get())
+	klog.InfoS("Version info", "version", version.Get())
 
 	// TODO(vmarmol): Use container config for this.
 	var oomAdjuster *oom.OOMAdjuster
 	if s.OOMScoreAdj != nil {
 		oomAdjuster = oom.NewOOMAdjuster()
 		if err := oomAdjuster.ApplyOOMScoreAdj(0, int(*s.OOMScoreAdj)); err != nil {
-			klog.V(2).Info(err)
+			klog.V(2).ErrorS(err, "failed to apply OOMScore")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:655]: Maybe Info ->InfoS ?

File: cmd/kube-proxy/app/server.go [hunk 16]
```diff
@@ -786,10 +790,10 @@ func getConntrackMax(config kubeproxyconfig.KubeProxyConntrackConfiguration) (in
 		}
 		scaled := int(*config.MaxPerCore) * goruntime.NumCPU()
 		if scaled > floor {
-			klog.V(3).Infof("getConntrackMax: using scaled conntrack-max-per-core")
+			klog.V(3).InfoS("getConntrackMax: using scaled conntrack-max-per-core")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:793]:
```suggestion
			klog.V(3).InfoS("GetConntrackMax: using scaled conntrack-max-per-core")
```

File: cmd/kube-proxy/app/server.go [hunk 17]
```diff
@@ -786,10 +790,10 @@ func getConntrackMax(config kubeproxyconfig.KubeProxyConntrackConfiguration) (in
...
 			return scaled, nil
 		}
-		klog.V(3).Infof("getConntrackMax: using conntrack-min")
+		klog.V(3).InfoS("getConntrackMax: using conntrack-min")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:796]:
```suggestion
		klog.V(3).InfoS("GetConntrackMax: using conntrack-min")
```

File: cmd/kube-proxy/app/server_others.go [hunk 18]
```diff
@@ -191,7 +191,7 @@ func newProxyServer(
 			// Just because the feature gate is enabled doesn't mean the node
 			// actually supports dual-stack
 			if _, err := ipt[1].ChainExists(utiliptables.TableNAT, utiliptables.ChainPostrouting); err != nil {
-				klog.Warningf("No iptables support for IPv6: %v", err)
+				klog.ErrorS(err, "No iptables support for IPv6")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:194]: Maybe Warningf ->InfoS
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

File: cmd/kube-proxy/app/server_others.go [hunk 19]
```diff
@@ -200,20 +200,20 @@ func newProxyServer(
 		}
 	}
 	if dualStack {
-		klog.V(0).Infof("kube-proxy running in dual-stack mode, %s-primary", iptInterface.Protocol())
+		klog.V(0).InfoS("kube-proxy running in dual-stack mode", "IP family", iptInterface.Protocol())
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:203]:
```suggestion
		klog.V(0).InfoS("Kube-proxy running in dual-stack mode", "IP family", iptInterface.Protocol())
```

File: cmd/kube-proxy/app/server_others.go [hunk 20]
```diff
@@ -200,20 +200,20 @@ func newProxyServer(
...
 	} else {
-		klog.V(0).Infof("kube-proxy running in single-stack %s mode", iptInterface.Protocol())
+		klog.V(0).InfoS("kube-proxy running in single-stack mode", "IP family", iptInterface.Protocol())
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:205]:
```suggestion
		klog.V(0).InfoS("Kube-proxy running in single-stack mode", "IP family", iptInterface.Protocol())
```

File: cmd/kube-proxy/app/server_others.go [hunk 21]
```diff
@@ -268,9 +268,9 @@ func newProxyServer(
 		}
 		proxymetrics.RegisterMetrics()
 	} else if proxyMode == proxyModeIPVS {
-		klog.V(0).Info("Using ipvs Proxier.")
+		klog.V(0).InfoS("Using ipvs Proxier.")
 		if dualStack {
-			klog.V(0).Info("creating dualStackProxier for ipvs.")
+			klog.V(0).InfoS("creating dualStackProxier for ipvs.")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:273]:
```suggestion
			klog.V(0).InfoS("Creating dualStackProxier for ipvs.")
```

File: cmd/kube-proxy/app/server_others.go [hunk 22]
```diff
@@ -454,18 +454,18 @@ func getLocalDetector(mode proxyconfigapi.LocalMode, config *proxyconfigapi.Kube
 	switch mode {
 	case proxyconfigapi.LocalModeClusterCIDR:
 		if len(strings.TrimSpace(config.ClusterCIDR)) == 0 {
-			klog.Warning("detect-local-mode set to ClusterCIDR, but no cluster CIDR defined")
+			klog.InfoS("detect-local-mode set to ClusterCIDR, but no cluster CIDR defined")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:457]:
```suggestion
			klog.InfoS("Detect-local-mode set to ClusterCIDR, but no cluster CIDR defined")
```

File: cmd/kube-proxy/app/server_others.go [hunk 23]
```diff
@@ -454,18 +454,18 @@ func getLocalDetector(mode proxyconfigapi.LocalMode, config *proxyconfigapi.Kube
...
 			break
 		}
 		return proxyutiliptables.NewDetectLocalByCIDR(config.ClusterCIDR, ipt)
 	case proxyconfigapi.LocalModeNodeCIDR:
 		if len(strings.TrimSpace(nodeInfo.Spec.PodCIDR)) == 0 {
-			klog.Warning("detect-local-mode set to NodeCIDR, but no PodCIDR defined at node")
+			klog.InfoS("detect-local-mode set to NodeCIDR, but no PodCIDR defined at node")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:463]:
```suggestion
			klog.InfoS("Detect-local-mode set to NodeCIDR, but no PodCIDR defined at node")
```

File: cmd/kube-proxy/app/server_others.go [hunk 24]
```diff
@@ -475,14 +475,14 @@ func getDualStackLocalDetectorTuple(mode proxyconfigapi.LocalMode, config *proxy
 	switch mode {
 	case proxyconfigapi.LocalModeClusterCIDR:
 		if len(strings.TrimSpace(config.ClusterCIDR)) == 0 {
-			klog.Warning("detect-local-mode set to ClusterCIDR, but no cluster CIDR defined")
+			klog.InfoS("detect-local-mode set to ClusterCIDR, but no cluster CIDR defined")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:478]:
```suggestion
			klog.InfoS("Detect-local-mode set to ClusterCIDR, but no cluster CIDR defined")
```

File: cmd/kube-proxy/app/server_others.go [hunk 25]
```diff
@@ -475,14 +475,14 @@ func getDualStackLocalDetectorTuple(mode proxyconfigapi.LocalMode, config *proxy
...
 			break
 		}
 
 		clusterCIDRs := cidrTuple(config.ClusterCIDR)
 
 		if len(strings.TrimSpace(clusterCIDRs[0])) == 0 {
-			klog.Warning("detect-local-mode set to ClusterCIDR, but no IPv4 cluster CIDR defined, defaulting to no-op detect-local for IPv4")
+			klog.InfoS("detect-local-mode set to ClusterCIDR, but no IPv4 cluster CIDR defined, defaulting to no-op detect-local for IPv4")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:485]:
```suggestion
			klog.InfoS("Detect-local-mode set to ClusterCIDR, but no IPv4 cluster CIDR defined, defaulting to no-op detect-local for IPv4")
```

File: cmd/kube-proxy/app/server_others.go [hunk 26]
```diff
@@ -491,14 +491,14 @@ func getDualStackLocalDetectorTuple(mode proxyconfigapi.LocalMode, config *proxy
 		}
 
 		if len(strings.TrimSpace(clusterCIDRs[1])) == 0 {
-			klog.Warning("detect-local-mode set to ClusterCIDR, but no IPv6 cluster CIDR defined, , defaulting to no-op detect-local for IPv6")
+			klog.InfoS("detect-local-mode set to ClusterCIDR, but no IPv6 cluster CIDR defined, , defaulting to no-op detect-local for IPv6")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:494]:
```suggestion
			klog.InfoS("Detect-local-mode set to ClusterCIDR, but no IPv6 cluster CIDR defined, , defaulting to no-op detect-local for IPv6")
```

**Comment by yxxhero** [comment]
@yangjunmyfm192085 Thanks your reply. I will fix them.

File: cmd/kube-proxy/app/server.go (hunk 6 above)
Comment (yxxhero) [reply cmd/kube-proxy/app/server.go:480]: https://pkg.go.dev/k8s.io/klog@v0.3.2#Fatalf     it is os.Exit(255),  is it ok? Thank very much.

Comment (yangjunmyfm192085) [reply cmd/kube-proxy/app/server.go:480]: /cc @ehashman

Comment (yangjunmyfm192085) [reply cmd/kube-proxy/app/server.go:480]: Maybe this is helpful
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

Comment (yxxhero) [reply cmd/kube-proxy/app/server.go]: @yangjunmyfm192085  Thank you for your guidance, I have fixed it. Thanks very much again.

**Comment by yxxhero** [comment]
@yangjunmyfm192085  @Cweiping Looking forward to your reply. Thanks very much.

**Comment by yxxhero** [comment]
any update? What shuld I do next?

**Comment by yxxhero** [comment]
@yangjunmyfm192085 thanks very much.

**Comment by yxxhero** [comment]
any update?

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm

**Review by yangjunmyfm192085 — COMMENTED** [review]
need rebase

**Comment by yxxhero** [comment]
@yangjunmyfm192085 rebase is ok.

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm

File: cmd/kube-proxy/app/conntrack.go (hunk 1 above)
Comment (aojea) [reply cmd/kube-proxy/app/conntrack.go:100]: why is this marked as resolved if the suggestion wasn't accepted?
which one is the preferred?

Comment (aojea) [reply cmd/kube-proxy/app/conntrack.go:100]: the kep seems to prefer the keysAndValues format

File: cmd/kube-proxy/app/conntrack.go (hunk 2 above)
Comment (aojea) [reply cmd/kube-proxy/app/conntrack.go:115]: the kep seems to prefer caps too https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#new-klog-methods

**Comment by aojea** [comment]
I'm not very familiar with the structured logging, but I found confusing and inconsistent the use of caps in some messages and the key values
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#new-klog-methods

File: cmd/kube-proxy/app/conntrack.go (hunk 1 above)
Comment (yangjunmyfm192085) [reply cmd/kube-proxy/app/conntrack.go:100]: ouch! Could you take a look again,@yxxhero?

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm cancel
could you adopt the suggestion of @aojea

**Comment by yxxhero** [comment]
@yangjunmyfm192085 yes, I will fix.

**Comment by yxxhero** [comment]
@aojea @yangjunmyfm192085 I am so sorry. The rebase operation lost the suggested modification information. Thanks very much.

**Review by yangjunmyfm192085 — COMMENTED** [review]
/test pull-kubernetes-integration

File: cmd/kube-proxy/app/server.go [hunk 27]
```diff
@@ -221,7 +221,7 @@ func NewOptions() *Options {
 // Complete completes all the required options.
 func (o *Options) Complete() error {
 	if len(o.ConfigFile) == 0 && len(o.WriteConfigTo) == 0 {
-		klog.Warning("WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.")
+		klog.InfoS("WARNING: all flags other than --config, --write-config-to, and --cleanup are deprecated. Please begin using a config file ASAP.")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:224]: Maybe we should remove the period.

File: cmd/kube-proxy/app/server.go [hunk 28]
```diff
@@ -432,7 +432,7 @@ func (o *Options) loadConfig(data []byte) (*kubeproxyconfig.KubeProxyConfigurati
 		}
 
 		// Continue with the v1alpha1 object that was decoded leniently, but emit a warning.
-		klog.Warningf("using lenient decoding as strict decoding failed: %v", err)
+		klog.InfoS("Using lenient decoding as strict decoding failed", "error", err)
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:435]:
```suggestion
		klog.InfoS("Using lenient decoding as strict decoding failed", "err", err)
```

File: cmd/kube-proxy/app/server.go [hunk 29]
```diff
@@ -550,7 +554,7 @@ func createClients(config componentbaseconfig.ClientConnectionConfiguration, mas
 	var err error
 
 	if len(config.Kubeconfig) == 0 && len(masterOverride) == 0 {
-		klog.Info("Neither kubeconfig file nor master URL was specified. Falling back to in-cluster config.")
+		klog.InfoS("Neither kubeconfig file nor master URL was specified. Falling back to in-cluster config.")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:557]:
```suggestion
		klog.InfoS("Neither kubeconfig file nor master URL was specified. Falling back to in-cluster config")
```

File: cmd/kube-proxy/app/server.go [hunk 30]
```diff
@@ -646,14 +650,14 @@ func serveMetrics(bindAddress, proxyMode string, enableProfiling bool, errCh cha
...
-	klog.Infof("Version: %+v", version.Get())
+	klog.InfoS("Version info", "version", version.Get())
 
 	// TODO(vmarmol): Use container config for this.
 	var oomAdjuster *oom.OOMAdjuster
 	if s.OOMScoreAdj != nil {
 		oomAdjuster = oom.NewOOMAdjuster()
 		if err := oomAdjuster.ApplyOOMScoreAdj(0, int(*s.OOMScoreAdj)); err != nil {
-			klog.V(2).Info(err)
+			klog.V(2).InfoS("Failed to apply OOMScore", "error", err)
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server.go:660]:
```suggestion
			klog.V(2).InfoS("Failed to apply OOMScore", "err", err)
```

File: cmd/kube-proxy/app/server_others.go [hunk 31]
```diff
@@ -137,7 +137,7 @@ func newProxyServer(
 	}
 
 	nodeIP := detectNodeIP(client, hostname, config.BindAddress)
-	klog.Infof("Detected node IP %s", nodeIP.String())
+	klog.InfoS("Detected node IP %s", "address", nodeIP.String())
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:140]:
```suggestion
	klog.InfoS("Detected node IP", "address", nodeIP.String())
```

File: cmd/kube-proxy/app/server_others.go [hunk 32]
```diff
@@ -209,14 +209,14 @@ func newProxyServer(
 	}
 
 	if proxyMode == proxyModeIPTables {
-		klog.V(0).Info("Using iptables Proxier.")
+		klog.V(0).InfoS("Using iptables Proxier.")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:212]:
```suggestion
		klog.V(0).InfoS("Using iptables Proxier")
```

File: cmd/kube-proxy/app/server_others.go [hunk 33]
```diff
@@ -209,14 +209,14 @@ func newProxyServer(
...
 		if config.IPTables.MasqueradeBit == nil {
 			// MasqueradeBit must be specified or defaulted.
 			return nil, fmt.Errorf("unable to read IPTables MasqueradeBit from config")
 		}
 
 		if dualStack {
-			klog.V(0).Info("creating dualStackProxier for iptables.")
+			klog.V(0).InfoS("Creating dualStackProxier for iptables.")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:219]:
```suggestion
			klog.V(0).InfoS("Creating dualStackProxier for iptables")
```

File: cmd/kube-proxy/app/server_others.go [hunk 34]
```diff
@@ -271,9 +271,9 @@ func newProxyServer(
 		}
 		proxymetrics.RegisterMetrics()
 	} else if proxyMode == proxyModeIPVS {
-		klog.V(0).Info("Using ipvs Proxier.")
+		klog.V(0).InfoS("Using ipvs Proxier.")
 		if dualStack {
-			klog.V(0).Info("creating dualStackProxier for ipvs.")
+			klog.V(0).InfoS("Creating dualStackProxier for ipvs.")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:276]: ditto

File: cmd/kube-proxy/app/server_others.go [hunk 35]
```diff
@@ -345,7 +345,7 @@ func newProxyServer(
 		}
 		proxymetrics.RegisterMetrics()
 	} else {
-		klog.V(0).Info("Using userspace Proxier.")
+		klog.V(0).InfoS("Using userspace Proxier.")
```
Comment (yangjunmyfm192085) [inline cmd/kube-proxy/app/server_others.go:348]: ditto

File: cmd/kube-proxy/app/server.go [hunk 36]
```diff
@@ -507,7 +510,8 @@ with the apiserver API to configure the proxy.`,
 	var err error
 	opts.config, err = opts.ApplyDefaults(opts.config)
 	if err != nil {
-		klog.Fatalf("unable to create flag defaults: %v", err)
+		klog.ErrorS(err, "Unable to create flag defaults")
+		os.Exit(1)
```
Comment (aojea) [inline cmd/kube-proxy/app/server.go:514]: what is the Fatalf equivalence with the logging struct change?
https://github.com/kubernetes/klog/blob/79f9f6b8291d38aaf33a0998e65a5bc254d51866/klog.go#L1511
we should have an unique way to translate these things for consistency
```
// Fatalf logs to the FATAL, ERROR, WARNING, and INFO logs,
// including a stack trace of all running goroutines, then calls os.Exit(255).
// Arguments are handled in the manner of fmt.Printf; a newline is appended if missing.
func Fatalf(format string, args ...interface{}) {
	logging.printf(fatalLog, logging.logr, logging.filter, format, args...)
}
```

Comment (yangjunmyfm192085) [reply cmd/kube-proxy/app/server.go:514]: yeah, we refer to
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md.
@yxxhero, In `user-facing change`, you should add `ACTION REQUIRED: Exit code changed from 255 to 1`

Comment (aojea) [reply cmd/kube-proxy/app/server.go:514]: perfect, thanks :+1:

File: cmd/kube-proxy/app/server_others.go [hunk 37]
```diff
@@ -271,9 +271,9 @@ func newProxyServer(
 		}
 		proxymetrics.RegisterMetrics()
 	} else if proxyMode == proxyModeIPVS {
-		klog.V(0).Info("Using ipvs Proxier.")
+		klog.V(0).InfoS("Using ipvs Proxier.")
```
Comment (aojea) [inline cmd/kube-proxy/app/server_others.go:274]: you've removed all trailing dots but this :smile:

**Comment by aojea** [comment]
/lgtm
/approve
just a nit missing

File: cmd/kube-proxy/app/server_others.go (hunk 37 above)
Comment (yxxhero) [reply cmd/kube-proxy/app/server_others.go:274]: In pursuit of perfection, I will fix it. Thanks very much.

**Comment by yxxhero** [comment]
@aojea  Looking forward to your review. Thanks very much.

**Comment by aojea** [comment]
/lgtm
Thanks
However, I don't know if this will qualify for 1.22, I'm not familiar with the structured logging implementation plans

**Comment by yxxhero** [comment]
> /retest
> This bot automatically retries jobs that failed/flaked on approved PRs (send feedback to [fejta](https://github.com/fejta)).
> 
> Review the [full test history](https://prow.k8s.io/pr-history/?org=kubernetes&repo=kubernetes&pr=98913) for this PR.
> 
> Silence the bot with an `/lgtm cancel` or `/hold` comment for consistent failures.

@aojea Thanks very much again. this issue require /hold comment?

**Comment by aojea** [comment]
> @aojea Thanks very much again. this issue require /hold comment?

No, that's the bot that retried a failing job.
Currently kubernetes is in code freeze, The PR will not be able to merge until 1.22 is released. 
Is okay , we should just wait a few weeks until master opens again

**Comment by yxxhero** [comment]
@aojea Thanks very much again.

**Pull request merged**